pkg/client/perseshttp/client.go

```diff
@@ -102,7 +103,23 @@ type RESTClient struct {
 	// base is the root URL for all invocations of the client
 	BaseURL *url.URL
 	// Set specific behavior of the client.  If not set http.DefaultClient will be used.
-	Client *http.Client
+	Client    *http.Client
+	authMutex sync.RWMutex
+}
+
+func (c *RESTClient) SetBasicAuth(username, password string) {
+	c.authMutex.Lock()
+	c.basicAuth = &secret.BasicAuth{
+		Username: username,
+		Password: password,
+	}
+	c.authMutex.Unlock()
```

Comment:
any reason to put the unlock at the end in these setters, vs in a defer in newRequest()?

Reply:
that's because here there is no return and it's a linear code. So there is only one way to leave it.

In the newRequest, there is something to return, so defer is necessary